app/src/main/kotlin/com/wire/android/ui/WireActivity.kt

```diff
@@ -50,31 +43,37 @@ class WireActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         installSplashScreen()
         super.onCreate(savedInstanceState)
-        handleDeepLink(intent)
-        setComposableContent()
-    }
-
-    private fun handleDeepLink(intent: Intent) {
         viewModel.handleDeepLink(intent)
+        setComposableContent()
     }
 
     override fun onNewIntent(intent: Intent?) {
-        intent?.let {
+        if (viewModel.handleDeepLinkOnNewIntent(intent)) {
             recreate()
-            viewModel.handleDeepLink(intent)
         }
         super.onNewIntent(intent)
     }
 
+    override fun onResume() {
+        super.onResume()
+        viewModel.activityOnResume()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        viewModel.activityOnPause()
+    }
```

Comment:
Just FYI - it can be done by implementing LifecycleObserver in ViewModel and just adding observer in Activity 🙂  
This way we can listen for all states in ViewModel without need to override all these functions and calling viewmodel inside them 😄 

You just do this in ViewModel:
```
public class MyViewModel extends ViewModel implements LifecycleObserver {
  @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
  protected void onLifeCyclePause() {
      // do something
  }
}
```
and in Activity:
```
lifecycle.addObserver(myViewModel)
```

Reply:
damn! I knew i forgot about something :) thanks